core/src/Kokkos_HostSpace.hpp

```diff
@@ -138,39 +138,47 @@ namespace Kokkos {
 
 namespace Impl {
 
-static_assert(Kokkos::Impl::MemorySpaceAccess<Kokkos::HostSpace,
-                                              Kokkos::HostSpace>::assignable);
-
-template <typename S>
+template <class S>
 struct HostMirror {
  private:
+  using is_space = typename Kokkos::is_space<S>;
+  static_assert(is_space());
```

Comment:
```suggestion
  static_assert(is_space::value);
```
I don't really like the reliance on implicit conversion to bool here.

Reply:
We have an explicit `( )` operator which returns `value`.